tests/test_schema_errors/test_arrays.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+
+from openapi_tester import DocumentationError, SchemaTester
+
+example_schema_array = {"type": "array", "items": {"type": "string"}}
+example_array = ["string"]
+
+tester = SchemaTester()
+
+
+def test_nothing_wrong():
+    """ This should always pass """
+    tester.test_schema_section(example_schema_array, example_array, "")
+
+
+def test_empty_list():
+    """ An empty array should always pass """
+    tester.test_schema_section(example_schema_array, [], "")
+
+
+def test_null():
+    """ A null value should always raise an error """
```

Comment:
it shouldnt, not if it has null: True, x-nullable etc.

Reply:
Yep, but we're explicitly passing a schema element where it's not nullable. I originally named the test, `test_null_when_not_nullable`, but the next test is already called `test_nullable`, so I shortened it.

Reply:
I'll consolidate the two tests to this, so it's more clear

```python
def test_nullable():
    # A null value should always raise an error
    with pytest.raises(DocumentationError, match="Mismatched types, expected list but received NoneType"):
        tester.test_schema_section(example_schema_array, None, reference="")
    # Unless the schema specifies it should be nullable
    schema = {"type": "array", "items": {"type": "string"}, "nullable": True}
    tester.test_schema_section(schema, None, reference="")
```

Reply:
yup, is better IMO.